chips/sam4l/src/usbc/common_register.rs

```diff
@@ -0,0 +1,244 @@
+#![allow(dead_code)]
```

Comment:
This seems a bit like re-inventing `VolatileCell`. Can you explain the purpose of creating a new type rather than extending `VolatileCell` with the extra functionality?